OpenSim/Common/Component.cpp

```diff
@@ -1035,6 +1033,25 @@ void Component::setNextSubcomponentInSystem(const Component& sub) const
     }
 }
 
+void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
+{
+    if (versionNumber < XMLDocument::getLatestVersion()) {
+        if (versionNumber < 30508) {
+            // Here's an example of the change this function might make:
+            // Previous: <connectors>
+            //               <Connector_PhysicalFrame_ name="parent">
+            //                   <connectee_name>...</connectee_name>
+            //               </Connector_PhysicalFrame_>
+            //           </connectors>
+            // New:      <connector_parent_connectee_name>...
+            //               </connector_parent_connectee_name>
+            //
```

Comment:
>We can also ensure that the error message obtained when specifying a path to an object of the wrong type is very clear. For example, "The component at path '/foo/bar' is of type Ground, but must be of type Body." (I didn't check the code yet to see if such an error message is already in place).

I think that would be very helpful. One can turn to the *XMLBrowser* in the GUI or the *Doxygen* to learn of the type in the few instances that the comments are wiped out.  So, I am OK with this change.